omod/src/main/java/org/openmrs/web/controller/PortletController.java

```diff
@@ -195,43 +206,68 @@ public ModelAndView handleRequest(HttpServletRequest request, HttpServletRespons
 					}
 					
 					if (Context.hasPrivilege(PrivilegeConstants.GET_OBS)) {
-						List<Obs> patientObs = Context.getObsService().getObservationsByPerson(p);
-						model.put("patientObs", patientObs);
+						Integer limit = getLimitParameter(request, as, "dashboard.observations.maximumNumberToShow");
+						Integer startIndex = getStartIndexParameter(request);
+						
+						Person person = (Person) p;
+						List<Person> persons = Collections.singletonList(person);
+						
+						// Get most recent observations using limit parameter
+						List<Obs> paginatedObs = Context.getObsService().getObservations(persons, null, null, null, null, null, 
+						Collections.singletonList("obsDatetime desc"), limit, startIndex, null, null, false);					
+						
+						model.put("limit", limit);
```

Comment:
Can you remind me of why we are storing `limit` in the model?

Reply:
We storing `limit` in the model to make it accessible to/for the JSP and also keeping the pagination logic synchronised between the backend and JSP frontend

Reply:
The [patientEncounters](https://github.com/openmrs/openmrs-module-legacyui/blob/89cdd965a1acf1c04909186edd78950a19fae668/omod/src/main/webapp/portlets/patientEncounters.jsp#L4) jsp

Reply:
Do you mind pointing me to the actual line that uses limit?

Reply:
Looking at the dashboard JSP file, I don't actually see any direct reference to a limit or pagination for observations, safe to assume removing it. If the view needs the page size, it can derive it from patientObs.size()